src/swh/parser.js

```diff
@@ -34,31 +34,39 @@ class Parser {
    */
   advance() {
     this.tokIdx++;
-    if (this.tokIdx < this.tokens.length) {
-      this.currentTok = this.tokens[this.tokIdx];
-    }
+    this.updateCurrentTok();
     return this.currentTok;
   }
 
   /**
-   * moves to the previous token from the lexer
+   * moves backwards through the tokens for a given number of steps
+   * @param {Number} amount number of advancements to reverse
```

Comment:
According to jsdoc, you need to provide a name for @param {}. Is that not necessary?